k9mail/src/main/java/com/fsck/k9/notification/NewMailNotifications.java

```diff
@@ -0,0 +1,166 @@
+package com.fsck.k9.notification;
+
+
+import android.app.Notification;
+import android.support.v4.app.NotificationManagerCompat;
+import android.util.SparseArray;
+
+import com.fsck.k9.Account;
+import com.fsck.k9.K9;
+import com.fsck.k9.K9.NotificationHideSubject;
+import com.fsck.k9.activity.MessageReference;
+import com.fsck.k9.mailstore.LocalMessage;
+
+
+class NewMailNotifications {
+    private final NotificationController controller;
+    private final NotificationContentCreator contentCreator;
+    private final DeviceNotifications deviceNotifications;
+    private final WearNotifications wearNotifications;
+    private final SparseArray<NotificationsHolder> notifications = new SparseArray<NotificationsHolder>();
+    private final Object lock = new Object();
```

Comment:
is this just a pattern for clarity? because it's equivalent to synchronizing over `this`